changelogs/fragments/8542-fix-proxmox-volume-handling.yml

```diff
@@ -0,0 +1,5 @@
+bugfixes:
+  - proxmox - fix idempotency on creation of mount volumes using Proxmox' special ``<storage>:<size>`` syntax (https://github.com/ansible-collections/community.general/issues/8407, https://github.com/ansible-collections/community.general/pull/8542).
+minor_changes:
+  - proxmox - add ``disk_volume`` and ``mount_volumes`` keys for better readability (https://github.com/ansible-collections/community.general/pull/8542).
+  - proxmox - translate the old ``disk`` and ``mounts`` keys to the new handling internally (https://github.com/ansible-collections/community.general/pull/8542).
```

Comment:
I believe you should break this PR into two parts: one for the bugfix, the other for the new features. The bugfix gets to be backported to previous versions, while features do not. 
@felixfontein please confirm

Reply:
Currently, those changes are intertwined as I first convert the old syntax to the new one and then do the idempotency checks. Breaking this PR apart would consist of just removing the ability to *use* the new keywords; I'm not sure if that's worth it.

Also, some people *may* use this bug, though then it's a question of whether we want to support that behavior.

Reply:
I think it's OK to keep bugfix and feaure combined here. (In general it's better to separate them.)

> Also, some people *may* use this bug

https://xkcd.com/1172/ :)

Reply:
Fair enough :-)